### PR TITLE
Create 0022-submit-domain-request-user-flow

### LINKED ISSUE #800: Add an ADR for decision around Submit Domain Request User Flow
### Issue Description

Following the ADR process (https://github.com/cisagov/getgov/tree/main/docs/architecture/decisions), add an ADR to codify the decision on how users are permitted or unpermitted from submitting an application.

### Additional Context (optional)

There was rounds of discussion after thinking through the implications of Bug #394 . We captured those discussions and came to a decision as a group. Now it's time to record that decision formally.

### Issue Links

🔄 Relates to: #394

### PR DESCRIPTION
**Pull request by gabydisarli**
## 🗣 Description ##

resolves #800

### CODE REVIEW

File: docs/architecture/decisions/0022-submit-domain-request-user-flow
Comment (abroddrick) [inline docs/architecture/decisions/0022-submit-domain-request-user-flow:15]: I would add here that the user can also just have an existing domain, instead of an approved application. You mention this in the decision, so good to have it here too.

File: docs/architecture/decisions/0022-submit-domain-request-user-flow
Comment (abroddrick) [inline docs/architecture/decisions/0022-submit-domain-request-user-flow:21]: The link here looked weird, and I realized this file isn't showing as a .md file. Can you add the file extension ".md", that should make the link show right I think. The other ADRS look like they are also all .md files.

**Review by abroddrick — APPROVED** [review]
A couple minor changes, but I am good with this ADR. I would say wait on @h-m-f-t to approve it too before merging though.

File: docs/architecture/decisions/0022-submit-domain-request-user-flow
Comment (h-m-f-t) [reply docs/architecture/decisions/0022-submit-domain-request-user-flow:21]: I like avoiding "here" and "this" as hyperlink anchors because doing so makes skimming easier. 

Consider revising this href:

> A user flow diagram that demonstrates this logic can be viewed at this link](https://miro.com/app/board/uXjVM3jz3Bs=/?share_link_id=875307531981)

to

> {new line}
> A [user flow diagram](https://miro.com/app/board/uXjVM3jz3Bs=/?share_link_id=875307531981) demonstrates our decision.

**Review by h-m-f-t — CHANGES_REQUESTED** [review]
Really excellent work preparing this! I have a few requested changes but this is in pretty great shape.

File: docs/architecture/decisions/0022-submit-domain-request-user-flow
Comment (gabydisarli) [reply docs/architecture/decisions/0022-submit-domain-request-user-flow]: I've revised it to your suggestion. Thank you for making it much more readable!

Comment (gabydisarli) [reply docs/architecture/decisions/0022-submit-domain-request-user-flow]: @abroddrick I've fixed the file path, I think it works now?

File: docs/architecture/decisions/0022-submit-domain-request-user-flow
Comment (gabydisarli) [reply docs/architecture/decisions/0022-submit-domain-request-user-flow]: You are much more eloquent with words than I! thank you for the revised copy.

**Review by Katherine-Osos — APPROVED** [review]

File: docs/architecture/decisions/0022-submit-domain-request-user-flow
Comment (h-m-f-t) [reply docs/architecture/decisions/0022-submit-domain-request-user-flow]: Editing is always easier than creation! Thanks for an excellent initial draft.

**Review by h-m-f-t — APPROVED** [review]

**Pull request merged**
